### ISSUE TITLE
HTTP tunnel (aka HTTP proxy)

### ISSUE THREAD

**Issue by QuAzI** [dbeaver/dbeaver#4445]
Does DBeaver can use HTTP connection as a proxy to access to DB?
Like this https://www2.navicat.com/manual/online_manual/en/navicat/win_manual/HTTPSettings.html

**Comment by serge-rider** [comment]
No, unfortunately we don't support HTTP tunnels at the moment.

**Comment by anhnnp** [comment]
I hope to have this function in the future

**Comment by StiJans** [comment]
This feature is really needed so far.

**Comment by felangga** [comment]
Yes, I needed this too

**Comment by garretttaco** [comment]
I would also like to request HTTP proxy support.

**Comment by manju015** [comment]
I would like  to request  HTTP/https proxy too

**Comment by ayipeiger** [comment]
Yes this is the most important feature that we need, that's why i still using navicat anymore.

**Comment by serge-rider** [comment]
Some of my thoughts on this.

It is clear how to implement this feature although it still requires quite a bit of coding.
Navicat, SQLyog, EMS and dbForge have this feature (very similar implementations). But it comes from ancient times when php hosting over ftp was very popular.

HTTP tunnel is an obvious and horrible security hole. Taking into consideration that MySQL user passwords are often not very secured, never change, etc.
I wonder how many people still use http tunneling over php. Nowadays VPS hosting became extremely cheap and SSH tunnels are way more secure.

Moving to 7.1. Please vote for this if you still think that it is a must-have feature.

**Comment by felangga** [comment]
Yes SSH is more secure, but I would like a http tunnel too on the next release

**Comment by QuAzI** [comment]
> HTTP tunnel is an obvious and horrible security hole. Taking into consideration that MySQL user passwords are often not very secured, never change, etc.
You always can hide your script by using random chars in path and nobody except you will not know path. And you still need to know DB login and pass. Also .htaccess with IP grants still actual. Not a problem. Also nobody will use this way if normal SSH access present. But sometimes you have no way and must use it

**Comment by adiditm** [comment]
Yes, http tunnel (implementation like EMS, RazorSQL, Navicat) is very important and needed for many projects. In my office SSH access only allowed via limited SSH tunnel box (like Mikrotik box), login to that box first then login to servers. So DBeaver cannot connect directly to our servers. Also in my projects that hosted in shared hosting, I have no SSH access.

Thanks

**Comment by Mate20x** [comment]
@serge-rider 

I really need http channel verification

Hope this feature can be added

**Comment by Mate20x** [comment]
@serge-rider

Where I work, ports 22 and 3306 are always closed

Work at home
Use Navicat to connect to mysql through http channel

very convenient

So the http channel is needed

**Comment by CarGod** [comment]
@serge-rider @tati-kru 

Hi, I noticed that the function of HTTP tunnel has been planned for March 2020, but this function is still not seen in the current latest version. Can this function be done? I really need this feature, thank you.

**Comment by coolspring1293** [comment]
Reminder, reminder, I really need this function, thank you!

**Comment by bimbo1989** [comment]
Bumping this issue, the feature would be very welcome :)

**Comment by iBenMoore** [comment]
I wanted to throw out there that as a workaround some jdbc drivers can have HTTP proxy options themselves you can set for connections.  For example, for Snowflake I was able to work through a corp proxy by including in the "Driver Properties" of the connection new user properties useProxy, proxyHost, and proxyPort.

**Comment by poisonbian** [comment]
https://github.com/poisonbian/hersql
If you want to access mysql according http proxy, maybe you can try this tool.
